packages/blade/src/components/ProgressBar/__tests__/ProgressBar.native.test.tsx

```diff
@@ -0,0 +1,152 @@
+  });
+
+  it('should render high contrast negative intent ProgressBar', () => {
+    const { toJSON } = renderWithTheme(
+      <ProgressBar intent="negative" contrast="high" value={20} />,
+    );
+    expect(toJSON()).toMatchSnapshot();
+  });
+
+  it('should render low contrast notice intent ProgressBar', () => {
+    const { toJSON } = renderWithTheme(<ProgressBar intent="notice" contrast="low" value={20} />);
+    expect(toJSON()).toMatchSnapshot();
+  });
+
+  it('should render high contrast notice intent ProgressBar', () => {
+    const { toJSON } = renderWithTheme(<ProgressBar intent="notice" contrast="high" value={20} />);
+    expect(toJSON()).toMatchSnapshot();
+  });
+
+  it('should render low contrast information intent ProgressBar', () => {
+    const { toJSON } = renderWithTheme(
+      <ProgressBar intent="information" contrast="low" value={20} />,
+    );
+    expect(toJSON()).toMatchSnapshot();
+  });
+
+  it('should render high contrast information intent ProgressBar', () => {
+    const { toJSON } = renderWithTheme(
+      <ProgressBar intent="information" contrast="high" value={20} />,
+    );
+    expect(toJSON()).toMatchSnapshot();
+  });
+
+  it('should render low contrast neutral intent ProgressBar', () => {
+    const { toJSON } = renderWithTheme(<ProgressBar intent="neutral" contrast="low" value={20} />);
+    expect(toJSON()).toMatchSnapshot();
+  });
+
+  it('should render high contrast neutral intent ProgressBar', () => {
+    const { toJSON } = renderWithTheme(<ProgressBar intent="neutral" contrast="high" value={20} />);
+    expect(toJSON()).toMatchSnapshot();
+  });
+
+  it('should have accessibilityLabel', () => {
+    const { getByRole } = renderWithTheme(
+      <ProgressBar label="Label" accessibilityLabel="Downloading" value={70} />,
+    );
+
+    const progressbar = getByRole('progressbar');
+    expect(progressbar.findByProps({ accessibilityLabel: 'Downloading' })).toBeTruthy();
+  });
+
+  it('should not have accessibility violations', async () => {
+    const { toJSON } = renderWithTheme(<ProgressBar accessibilityLabel="Downloading" value={70} />);
+    await assertAccessible(toJSON());
+  });
+
+  it('should throw an error when the variant is meter and isIndeterminate is set.', () => {
+    try {
+      // @ts-expect-error testing failure case when there is no icon or text passed
+      renderWithTheme(<ProgressBar variant="meter" isIndeterminate={true} />);
+    } catch (error: unknown) {
+      if (error instanceof Error) {
+        expect(error.message).toEqual(
+          `[Blade: ProgressBar]: Cannot set 'isIndeterminate' when 'variant' is 'meter'.`,
+        );
+      }
+    }
+  });
+
+  it('should throw an error when the variant is meter and hidePercentage is set.', () => {
+    try {
+      // @ts-expect-error testing failure case when there is no icon or text passed
+      renderWithTheme(<ProgressBar variant="meter" hidePercentage={true} />);
+    } catch (error: unknown) {
```

Comment:
Let's not use try catch while testing for errors. use the `.toThrow` method. 

Because if the error never happen the `catch` block will never trigger and jest won't even complain.